### PR TITLE
jb/fix-command-line-overwrites

### PR DESCRIPTION
**Pull request by kaneda**
- Use yargs config to safely combine parsed JSON/YAML with yargs args

### CODE REVIEW

File: src/cli/parse_args.js [hunk 1]
```diff
@@ -552,8 +552,8 @@ function parse_args (args) {
 			dir = path.dirname(file);
 			try {
 				let doc = JSON.parse(fs.readFileSync(file));
+				argv = require('yargs').config(doc).argv;
```
Comment (torra) [inline src/cli/parse_args.js]: can we not get rid of these `try{} catch(){}` blocks and use the `yargs` config cli option? ie here: https://github.com/cquotient/nemesys/pull/90/files#diff-0bbc1fff1ca6aea7fa727b5fefd5be13R53

Comment (kaneda) [reply src/cli/parse_args.js]: Reading the file could still fail (e.g., if someone passes nonsensical arguments), so why not try/catch and alert the user?

In terms of the where, we join the yargs generated in the line you mentioned everywhere that uses them (but that's not all places), e.g., it's used here https://github.com/cquotient/nemesys/pull/90/files#diff-0bbc1fff1ca6aea7fa727b5fefd5be13R81 but not here https://github.com/cquotient/nemesys/pull/90/files#diff-0bbc1fff1ca6aea7fa727b5fefd5be13R71 (though that could be a mistake).

If we're going to use the common args *everywhere* we could put it up there instead.

Comment (torra) [reply src/cli/parse_args.js]: yargs probably has similar handling of a bad input json. it's a good point about using common args everywhere - we do, in fact that is where the config file params are defined now anyways: https://github.com/cquotient/nemesys/blob/9da8d6e040850a282c09e6dd23d84f4d0258dcba/src/cli/parse_args.js#L38-L54

Comment (kaneda) [reply src/cli/parse_args.js]: In that case I'll add this to the one place it's missing and move this up @torra

Comment (kaneda) [reply src/cli/parse_args.js]: @torra forgot to reply to the other part of that, which is that we're the ones parsing the JSON/YAML in my PR still, and then we let yargs deal with the combination through the "config" function, so I think we should still try/catch here

Comment (kaneda) [reply src/cli/parse_args.js]: Thought about this some more. We're defining options in the common args, not processing commands, I don't think it's possible to move the file processing block to common args for this reason.

Comment (torra) [reply src/cli/parse_args.js]: can't we let yargs do the parsing? ie, can it be like this:

Comment (kaneda) [reply src/cli/parse_args.js]: I had no luck using "extends" ( http://yargs.js.org/docs/#api-configobject-extends-keyword ) which is what I think we'd need to do for this.

**Pull request merged**